recipes/ios-cmake/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+
+from conans import ConanFile, tools
+
+import os
+import platform
+import copy
+
+
+class IosCMakeConan(ConanFile):
+    name = "ios-cmake"
+    license = "BSD-3-Clause"
+    settings = "os" , "arch"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/leetal/ios-cmake"
+    options = {
+        "enable_bitcode": [True, False],
+        "enable_arc": [True, False],
+        "enable_visibility": [True, False],
+        "enable_strict_try_compile": [True, False],
+        "ios_target": ["OS", "OS64", "OS64COMBINED",
+                       "SIMULATOR", "SIMULATOR64",
+                       "TVOS", "TVOSCOMBINED",
+                       "SIMULATOR_TVOS", "WATCHOS",
+                       "WATCHOSCOMBINED", "SIMULATOR_WATCHOS"]
+    }
+    default_options = {
+        "enable_bitcode": False,
+        "enable_arc": False,
+        "enable_visibility": False,
+        "enable_strict_try_compile": False,
+        "ios_target": "OS64COMBINED", 
+    }
+    description = "ios Cmake toolchain to (cross) compile macOS/iOS/watchOS/tvOS"
+    url = "https://github.com/leetal/ios-cmake"
+    exports_sources =  "cmake-wrapper"
+
+    def _source_subfolder(self):
+        return os.path.join(self.source_folder, "source_subfolder")
+
```

Comment:
```suggestion

    def configure(self):
        if "armv7" not in str(self.settings.arch) or "armv8" not in str(self.settings.arch):
            raise ConanInvalidConfiguration("Architecture not supported")
        if tools.is_apple_os(self.settings.os):
            raise ConanInvalidConfiguration("This toolchain only supports Apple os'es")

```

Reply:
@a4z checking the arch seems be reasonable. Could you please update configure()? Besides that, it looks good.

Reply:
Sorry for the late feedback, but I was on other tasks, one of them vacation :-) 

I do not think that makes too much senes, if you use a wrong architecture, your project will not compile.
It is also much more archs then arm7 and arm8, its x86_64, arm7, arm7s, arm8, arm8e and maybe even more (tv os and watch).
this list might also grow (apple silicon) so I do not see a big value in doing so.

So this will add a lot of work, possible wrong checks, and permanent maintenance, for relatively low, or zero, functionality.
If you do not have the correct arch in the profile, or specified, no one can help you anyway. 

one possible change I started thinking about is mapping the _ios_target_ option to the os version, because  I do not think that fat binaries are a topic for conan at the moment, and using them needs some more things anyway.